### PR TITLE
Reply in kind 1

### PR DESCRIPTION
**Pull request by bob2402**
#403 

- reply button in message actions
- handle reply state change
- reply indicator in editor
- generate reply tags

### CODE REVIEW

File: app/UI/message-panel.tsx
Comment (BlowaterNostr) [inline app/UI/message-panel.tsx:47]: Good catch

File: app/UI/app_update.tsx
Comment (BlowaterNostr) [inline app/UI/app_update.tsx:653]: Maybe we can simplify this to just `e` reply and `p` tagged pubkey. The current logic is hard to follow.

**Review by BlowaterNostr — CHANGES_REQUESTED** [review]

File: app/UI/app_update.tsx
Comment (BlowaterNostr) [inline app/UI/app_update.tsx:603]: here we have UI event and nostr event. Rename `event` to `ui_event` to be more readable.

**Comment by BlowaterNostr** [comment]
Overall is fine. Let's wait until the next piece and merge together.

File: app/UI/message-list.tsx
Comment (BlowaterNostr) [inline app/UI/message-list.tsx:45]: If we can get the `NostrEvent` at thie level, passing an `NostrEvent` might be better. `string | NoteID` might be too small for this interface. Up to you.

File: app/UI/nav.tsx
Comment (BlowaterNostr) [inline app/UI/nav.tsx:71]: It's better to make
```
const SecondaryTextColor = `text-[#XXXXXX]`
```
in the colors file so that the caller code here does not need to prefix `text-`

**Review by BlowaterNostr — CHANGES_REQUESTED** [review]

**Review by BlowaterNostr — APPROVED** [review]

File: app/UI/nav.tsx
Comment (bob2402) [reply app/UI/nav.tsx:71]: Refactor colors separately later

**Pull request merged**
